ui/packages/app/vite.config.js

```diff
@@ -1,19 +1,27 @@
 import { defineConfig } from "vite";
 import { svelte } from "@sveltejs/vite-plugin-svelte";
 import sveltePreprocess from "svelte-preprocess";
+import path from "path";
+import fs from "fs";
 
 // this is dupe config, gonna try fix this
 import tailwind from "tailwindcss";
 import nested from "tailwindcss/nesting/index.js";
 
+const GRADIO_VERSION = process.env.GRADIO_VERSION;
+const CDN_URL = `https://gradio.s3-us-west-2.amazonaws.com/${GRADIO_VERSION}/`;
```

Comment:
FWIW S3 is not really a CDN ie. is pretty slow especially when you're far from the region (here us-west) but it'd be pretty trivial to hook a Cloudfront distribution in front of it if needed to speed things up @pngwn

Reply:
Yeah, I agree. After 3.0 I'd like to rethink our CDN stuff. It would be cool to have a nice `cdn.gradio.app` with `/` resolving to the js entrypoint (which is currently broken). Bare url would redirect to latest. `cdn.gradio.app/3/` redirecting to latest 3.x, `cdn.gradio.app/3.1/` redirecting to latest `3.1.x` etc. For users embedding gradio via CDN, needing to manually track versions is frustrating and people will get out of date quickly.

But I also want to change how we do releases (+ asset deployments). I'd like us to have everything in CI with the asset deployments IaCed with TF or something (i assume we use TF here but havent checked).

Reply:
Use of the phrase 'CDN' is very flexible here. :D But using a real CDN is the aim.